internal/util/update/update_test.go

```diff
@@ -703,6 +704,93 @@ func TestCommand_Run_failInvalidPath(t *testing.T) {
 	}
 }
 
+func TestCommand_Run_badUpstreamLock(t *testing.T) {
+	testCases := map[string]struct {
+		dir              string
+		commit           string
+		repo             string
+		expectedErrorMsg string
+	}{
+		"bad commit": {
+			dir:              "/",
+			commit:           "fake",
+			repo:             testutil.Upstream,
+			expectedErrorMsg: "unknown revision or path"},
+		"bad dir": {
+			dir:              "/fake",
+			commit:           "",
+			repo:             testutil.Upstream,
+			expectedErrorMsg: "no such file or directory"},
+		"bad repo": {
+			dir:              "/",
+			commit:           "",
+			repo:             "fake",
+			expectedErrorMsg: "does not appear to be a git repository"},
+	}
+	for tn, tc := range testCases {
+		t.Run(tn, func(t *testing.T) {
+			upstreamContent := testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithResource(pkgbuilder.DeploymentResource),
+				Branch: masterBranch,
+			}
+
+			// Using a TestSetupManager for just the remote repo to get a directory
+			// for use in the local UpstreamLock.
+			remote := &testutil.TestSetupManager{
```

Comment:
We should only need to create one `TestSetupManager`. It goes through the steps of creating all the repos (as defined by the `ReposChanges` property), set up a local workspace, and then get the package from the repo name `upstream` into the workspace. The `TestSetupManager` keeps references to the created repos and workspace so we can use them.

I think in this case you should be able to basically just combine the two instances into one.